src/routes/products/idSearcher.js

```diff
@@ -0,0 +1,35 @@
+const products = require("../../db/products/all-products.json");
+const url = require("url");
+
+const idSearcher = (request, response) => {
+  const urlObj = url.parse(request.url);
+  console.log("urlObj :", urlObj);
+  const idFromPath = getIdFromPath(urlObj.path);
+  const idProd = products.filter(el => el.id === Number(idFromPath));
+  let foundedProducts;
+
+  if (idProd.length === 0 || !idProd) {
+    foundedProducts = {
+      status: "no products",
+      products: []
+    };
+  } else {
+    foundedProducts = {
+      status: "success",
+      products: idProd
+    };
+  }
+
+  response.writeHead(200, { "Content-Type": "application/json" });
+  response.write(JSON.stringify(foundedProducts));
+  response.end();
+};
+
+const getIdFromPath = urlPath => {
+  const urlPathLastIndex = urlPath.lastIndexOf("/");
```

Comment:
not sure if it's good approach. You can have a lot of path parameters, which divided by `/` so it's better to fetch path parameters by it's order number in url also. For instance, we have `/users/<userId>/products/<productId>` and we need to get userId and productId from url. Better to create func like this `(url, ...paramsOrder) => params`d>/products/<productId>`